src/components/atoms/Button/Button.tsx

```diff
@@ -152,26 +87,16 @@ const SButton = styled.button<IButtonProps>`
 `;
 
 const Button: React.FC<IButtonProps> = props => {
-  const { children, leftIcon, rightIcon, ...rest } = props;
+  const { children, leftIcon, rightIcon, loading, styling = 'primary', ...rest } = props;
+  const isLoading = styling !== 'link' && loading;
 
   return (
-    <SButton {...rest}>
-      {leftIcon || rightIcon ? (
-        <>
-          {!!leftIcon && leftIcon}
-          <SText rightIcon={rightIcon} leftIcon={leftIcon}>
-            {children}
-          </SText>
-          {!!rightIcon && rightIcon}
-        </>
-      ) : (
-        children
-      )}
+    <SButton styling={styling} {...rest}>
+      {isLoading && <Spinner negative={styling === 'primary'} small />}
+      {'\u00A0 '}
```

Comment:
This is temporal to have _white-space_ between the spinner and the text.

Once a [`<Spacing />` component is made](https://jira.zopa/browse/ZDS-9) I'm happy to update here.